crates/router/src/db/relay.rs

```diff
@@ -35,6 +35,14 @@ pub trait RelayInterface {
         merchant_key_store: &domain::MerchantKeyStore,
         relay_id: &common_utils::id_type::RelayId,
     ) -> CustomResult<hyperswitch_domain_models::relay::Relay, errors::StorageError>;
+
+    async fn find_relay_by_connector_reference_id(
+        &self,
+        key_manager_state: &KeyManagerState,
+        merchant_key_store: &domain::MerchantKeyStore,
+        connector_reference_id: &str,
+        profile_id: &common_utils::id_type::ProfileId,
+    ) -> CustomResult<hyperswitch_domain_models::relay::Relay, errors::StorageError>;
```

Comment:
Same here:

```suggestion
    async fn find_relay_by_profile_id_connector_reference_id(
        &self,
        key_manager_state: &KeyManagerState,
        merchant_key_store: &domain::MerchantKeyStore,
        profile_id: &common_utils::id_type::ProfileId,
        connector_reference_id: &str,
    ) -> CustomResult<hyperswitch_domain_models::relay::Relay, errors::StorageError>;
```

And fix it in trait implementations and usages.